vars/rollingUpgradePipeline.groovy

```diff
@@ -170,6 +187,103 @@ def call(Map pipelineParams) {
                                                 }
                                             }
                                         }
+                                        stage('BYO Scylladb [optional]') {
+                                            catchError(stageResult: 'FAILURE') {
+                                                script {
+                                                    wrap([$class: 'BuildUser']) {
+                                                        dir('scylla-cluster-tests/scylla-pkg') {
+                                                            if ( params.byo_scylla_repo && params.byo_scylla_branch ) {
```

Comment:
can't we package this whole logic into it's own module, like the rest of the stage are implemented ?

Reply:
so we can reuse it in more pipelines, as needed ?

Reply:
If you look at the both approach for longevity and upgrades, you will see that they differ.
We can move it out, but it will be 2 modules, not one.

Reply:
yes having it as modules would be helpful, so it can be applied to more pipelines.

I don't want to see people start copying the whole thing from pipeline to pipeline.

Reply:
done